### PR TITLE
Fix local source cover updating

### PR DESCRIPTION
**Pull request by FooIbar**
Ref: tachiyomiorg/tachiyomi#10186

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaScreenModel.kt [hunk 1]
```diff
@@ -537,7 +538,13 @@ class MangaScreenModel(
...
         try {
             withIOContext {
-                val chapters = state.source.getChapterList(state.manga.toSManga())
+                val manga = state.manga.toSManga()
+                val chapters = state.source.getChapterList(manga)
+
+                // `LocalSource.getChapterList` has side effects since 82bdf63
+                if (state.manga != state.manga.copyFrom(manga)) {
+                    updateManga.awaitUpdateFromSource(state.manga, manga, manualFetch)
+                }
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaScreenModel.kt:547]: Since this is specific to LocalSource move the code inside it?

Comment (GameMaster007) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaScreenModel.kt:547]: Yes

Comment (FooIbar) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaScreenModel.kt]: It's not feasible as LocalSource doesn't (and shouldn't) have access to update manga states, have to do it in MangaScreenModel.

**Comment by FooIbar** [comment]
This patch attempts to fix https://git.mihon.dev/tachiyomi/tachiyomi/issues/8034#issuecomment-28275

**Pull request closed without merging**